tests/io/ini/test_util.py

```diff
@@ -3,92 +3,196 @@
-        values = self._create_valid_dummy_model_values()
-        values["numcoordinates"] = None
-
-        with pytest.raises(ValidationError):
-            TestCoordinatesValidator.DummyModel(**values)
-
-    def test_no_xcoordinates_given_while_expected_throws_value_error(self):
-        values = self._create_valid_dummy_model_values()
-        values["xcoordinates"] = None
-
-        with pytest.raises(ValidationError):
-            TestCoordinatesValidator.DummyModel(**values)
-
-    def test_no_ycoordinates_given_while_expected_throws_value_error(self):
-        values = self._create_valid_dummy_model_values()
-        values["ycoordinates"] = None
-
-        with pytest.raises(ValidationError):
-            TestCoordinatesValidator.DummyModel(**values)
-
-    def test_fewer_xcoordinates_than_expected_throws_value_error(self):
-        values = self._create_valid_dummy_model_values()
-        values["xcoordinates"] = [1, 2]
-
-        with pytest.raises(ValidationError):
-            TestCoordinatesValidator.DummyModel(**values)
-
-    def test_more_xcoordinates_than_expected_throws_value_error(self):
-        values = self._create_valid_dummy_model_values()
-        values["xcoordinates"] = [1, 2, 3, 4]
-
-        with pytest.raises(ValidationError):
-            TestCoordinatesValidator.DummyModel(**values)
-
-    def test_fewer_ycoordinates_than_expected_throws_value_error(self):
-        values = self._create_valid_dummy_model_values()
-        values["ycoordinates"] = [1, 2]
-
-        with pytest.raises(ValidationError):
-            TestCoordinatesValidator.DummyModel(**values)
-
-    def test_more_ycoordinates_than_expected_throws_value_error(self):
-        values = self._create_valid_dummy_model_values()
-        values["ycoordinates"] = [1, 2, 3, 4]
-
-        with pytest.raises(ValidationError):
-            TestCoordinatesValidator.DummyModel(**values)
-
-    def test_fewer_than_minimum_required_number_of_coordinates_throws_value_error(self):
-        values = self._create_valid_dummy_model_values()
-        values["numcoordinates"] = 1
-        values["xcoordinates"] = [1.23]
-        values["ycoordinates"] = [9.87]
-
-        with pytest.raises(ValidationError):
-            TestCoordinatesValidator.DummyModel(**values)
-
-    def _create_valid_dummy_model_values(self) -> Dict:
-        values = dict(
-            numcoordinates=3,
-            xcoordinates=[1.23, 4.56, 7.89],
-            ycoordinates=[9.87, 6.54, 3.21],
+    @pytest.mark.parametrize(
+        "values",
+        [
+            {},
+            {
+                "nodeid": "some_nodeid",
+                "branchid": "some_branchid",
+                "chainage": 1.23,
+                "xcoordinates": [4.56, 5.67, 6.78],
+                "ycoordinates": [7.89, 8.91, 9.12],
+                "numcoordinates": 3,
+            },
+            {
+                "xcoordinates": [4.56, 5.67, 6.78],
+                "ycoordinates": [7.89, 8.91, 9.12],
```

Comment:
Why is this one wrong? I thought we could only give coordinates?

Reply:
This one is wrong because `validate_num_coordinates` is True

Reply:
Aah, of course. Thank you.